### ISSUE TITLE
SafeParser fails to parse line

### ISSUE THREAD

**Issue by domderen** [postcss/postcss-safe-parser#7]
SafeParser has failed to parse this CSS contents:

```
.wrap .bluebutton {
	background: filter: progid:DXImageTransform.Microsoft.gradient(startColorStr='#2e9fbc', EndColorStr='#066791');
}
```

I know that `background: filter: ...` is not a correct usage, but it shouldn't crash the whole parsing process. Right now it fails with:

```
TypeError: Cannot read property '0' of undefined
    at SafeParser.decl (/proj/node_modules/postcss/lib/parser.js:203:25)
    at SafeParser.decl (/proj/node_modules/postcss-safe-parser/lib/safe-parser.js:60:36)
    at SafeParser.precheckMissedSemicolon (/proj/node_modules/postcss-safe-parser/lib/safe-parser.js:98:14)
    at SafeParser.decl (/proj/node_modules/postcss/lib/parser.js:236:14)
    at SafeParser.decl (/proj/node_modules/postcss-safe-parser/lib/safe-parser.js:60:36)
    at SafeParser.other (/proj/node_modules/postcss/lib/parser.js:135:30)
    at SafeParser.loop (/proj/node_modules/postcss/lib/parser.js:81:26)
    at safeParse (/proj/node_modules/postcss-safe-parser/lib/safe-parse.js:21:12)
    at new LazyResult (/proj/node_modules/postcss/lib/lazy-result.js:70:24)
    at Processor.process (/proj/node_modules/postcss/lib/processor.js:117:12)
```

Of if safe parser has to fail, it should throw an instance of [CssSyntaxError](http://api.postcss.org/CssSyntaxError.html) to be able to find the faulty line number.

**Comment by ai** [comment]
Thanks. I will try to fix it on this weekend.

**Comment by domderen** [comment]
@ai Thanks, that would be great. I'm also happy to help with a PR, but just wasn't sure of the direction. Should the parser try to handle this exception gracefully? For example should it be handled in `precheckMissedSemicolon`? Or should it catch the exception, and return it as [CssSyntaxError](http://api.postcss.org/CssSyntaxError.html)?

**Comment by ai** [comment]
Done 7c3e589

**Comment by ai** [comment]
SafeParser idea is never return `CssSyntaxError` :D. At least it should ignore the code.

**Comment by ai** [comment]
Released in 3.0.1

**Comment by domderen** [comment]
That was super fast, thanks! ;)
